From fc7db4dac0536128d73fe2c1f1cc0d668fc798cc Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Tue, 17 Mar 2020 21:58:37 -0700
Subject: [PATCH] CMake: Provide missing Findandroid_log.cmake

This CMake module is provided by the Hunter build system, so we need to
import this file for building without Hunter.
---
 cmake/CMakeLists.txt                |  3 +++
 cmake/modules/Findandroid_log.cmake | 32 +++++++++++++++++++++++++++++
 2 files changed, 35 insertions(+)
 create mode 100644 cmake/modules/Findandroid_log.cmake

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 8e5e68073..9b12514a6 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -19,6 +19,9 @@ endif()
 # Project
 project(protobuf C CXX)
 
+# Expose cmake modules for finding packages
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/modules)
+
 # Add c++11 flags
 if (CYGWIN)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
diff --git a/cmake/modules/Findandroid_log.cmake b/cmake/modules/Findandroid_log.cmake
new file mode 100644
index 000000000..67158b3a8
--- /dev/null
+++ b/cmake/modules/Findandroid_log.cmake
@@ -0,0 +1,32 @@
+#.rst:
+# Findandroid_log.cmake
+# ----------
+# Finds the Android log library. Based on a helper function in the Hunter
+# package manager.
+#
+# This will define the following imported target:
+#
+#   android_log::android_log - The Android log library
+#
+
+set(ANDROID_LOG_TARGET_NAME "android_log::android_log")
+
+find_library(ANDROID_LOG_LIBRARY "log")
+find_path(ANDROID_LOG_HEADER_DIR "android/log.h")
+
+if (NOT TARGET ${ANDROID_LOG_TARGET_NAME})
+  message(STATUS "Creating target '${ANDROID_LOG_TARGET_NAME}':")
+  message(STATUS "* library: '${ANDROID_LOG_LIBRARY}'")
+  message(STATUS "* header dir: '${ANDROID_LOG_HEADER_DIR}'")
+
+  add_library(${ANDROID_LOG_TARGET_NAME} UNKNOWN IMPORTED)
+  set_target_properties(
+    ${ANDROID_LOG_TARGET_NAME}
+    PROPERTIES
+    IMPORTED_LINK_INTERFACE_LANGUAGE "CXX"
+    IMPORTED_LOCATION "${ANDROID_LOG_LIBRARY}"
+    INTERFACE_INCLUDE_DIRECTORIES "${ANDROID_LOG_HEADER_DIR}"
+  )
+endif()
+
+mark_as_advanced(ANDROID_LOG_LIBRARY ANDROID_LOG_HEADER_DIR)
-- 
2.20.1

